tests/bluetooth/bsim/audio/src/tbs_test.c

```diff
@@ -69,6 +110,257 @@ static struct bt_conn_cb conn_callbacks = {
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not terminate call: %d\n", err);
+		return err;
+	}
+
+	printk("Retrieving call\n");
+	err = bt_tbs_retrieve(call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not retrieve call: %d\n", err);
+		return err;
+	}
+
+	printk("Terminating call\n");
+	err = bt_tbs_terminate(call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not terminate call: %d\n", err);
+		return err;
+	}
+
+	printk("Hold & retrieve test sucessfull\n");
+
+	return err;
+}
+
+static int test_join(void)
+{
+	int err;
+	uint8_t call_indexes[2];
+
+	printk("%s\n", __func__);
+	printk("Placing first call\n");
+	err = bt_tbs_originate(0, "tel:000000000001", &call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not originate first call: %d\n", err);
+		return err;
+	}
+
+	printk("Answering first call\n");
+	err = bt_tbs_remote_answer(call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not answer first call: %d\n", err);
+		return err;
+	}
+	printk("First call answered\n");
+
+	call_indexes[0] = (uint8_t)call_index;
+
+	printk("Placing second call\n");
+	err = bt_tbs_originate(0, "tel:000000000002", &call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not originate second call: %d\n", err);
+		return err;
+	}
+
+	printk("Answering second call\n");
+	err = bt_tbs_remote_answer(call_index);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not answer second call: %d\n", err);
+		return err;
+	}
+	printk("Second call answered\n");
+
+	call_indexes[1] = (uint8_t)call_index;
+
+	printk("Joining calls\n");
+	err = bt_tbs_join(2, call_indexes);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not join calls: %d\n", err);
+		return err;
+	}
+
+	err = bt_tbs_terminate(call_indexes[0]);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not terminate first call: %d\n", err);
+		return err;
+	}
+
+	err = bt_tbs_terminate(call_indexes[1]);
+	if (err != BT_TBS_RESULT_CODE_SUCCESS) {
+		FAIL("Could not terminate second call: %d\n", err);
+		return err;
+	}
+
+	printk("Join calls test succesfull\n");
+
+	return err;
+}
+
+static int test_tbs_server_only(void)
+{
+	int err = 0;
+
+	err = test_answer_terminate();
+	if (err) {
+		printk("Test answer failed\n");
+		return err;
+	}
+
+	err = test_hold_retrieve();
+	if (err) {
+		printk("Test hold & retrieve failed\n");
+		return err;
+	}
+
+	err = test_join();
+	if (err) {
+		printk("Test join failed\n");
+		return err;
+	}
+
+	err = test_provider_name();
+	if (err) {
+		printk("Test set provider name failed\n");
+		return err;
+	}
+
+	err = test_set_signal_strength();
+	if (err) {
+		printk("Test set signal strength failed\n");
+		return err;
+	}
+
+	err = test_set_bearer_technology();
+	if (err) {
+		printk("Test set bearer technology failed\n");
+		return err;
+	}
+
+	err = test_set_status_flags();
+	if (err) {
+		printk("Test set status flags failed\n");
+		return err;
+	}
```

Comment:
Since these are also calling `FAIL` on failure, this could be reduced to 

```suggestion
	test_answer_terminate();
	test_hold_retrieve();
	test_join();
	test_provider_name();
	test_set_signal_strength();
	test_set_bearer_technology();
	test_set_status_flags();
```